### PR TITLE
Redux signup integration

### LINKED ISSUE #1747: Integrating redux in Signup component
**Actual Behaviour**

Signup component lack the use of Redux as of now

**Expected Behaviour**

Signup component should also be integrated with redux

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AakashMallik**
Fixes #1747 

Changes: I have integrated redux into signup component: 
- [x] Added an action type. 
- [x] Defined an api function in api/index.js
- [x] Defined function in reducer to handle login

Demo Link: https://pr-1748-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by AakashMallik** [comment]
@akshatnitd Sir, there is just one problem, in our earlier implementation the status code told us which type of error was being faced by the app, but now we always get error code as -1. I tried to replicate your implementation in Login component and I found nothing that was handling status code. Let me know if you have some suggestions.

File: src/apis/index.js [hunk 1]
```diff
@@ -16,3 +16,10 @@ export function getLogin(payload) {
   const url = `${API_URL}/${AUTH_API_PREFIX}/login.json`;
   return ajax.get(url, { login: email, password, type: 'access-token' });
 }
+
+export function getSignup(payload) {
+  console.log('fired');
```
Comment (akshatnitd) [inline src/apis/index.js]: Remove `console.log`

File: src/redux/reducers/app.js [hunk 2]
```diff
@@ -18,13 +18,19 @@ export default handleActions(
       };
     },
     [actionTypes.APP_GET_LOGIN](state, { payload }) {
+      console.log('Login fired');
```
Comment (akshatnitd) [inline src/redux/reducers/app.js]: Remove `console.log`

File: src/components/Auth/SignUp/SignUp.react.js [hunk 3]
```diff
@@ -1,49 +1,85 @@
...
-      passwordError: true,
-      passwordConfirmError: true,
-      passwordValue: '',
-      confirmPasswordValue: '',
-      msg: '',
-      success: false,
-      open: false,
-      openLogin: false,
-      openForgotPassword: false,
+      emailError: false,
```
Comment (akshatnitd) [inline src/components/Auth/SignUp/SignUp.react.js]: Only 1 state out of `emailError` and `emailErrorMessage` would be sufficient. Use `emailErrorMessage === ''` for falsy values. Do the same for other state variables as well

File: src/components/Auth/SignUp/SignUp.react.js [hunk 4]
```diff
@@ -65,276 +101,200 @@ export default class SignUp extends Component {
...
         serverFieldError: false,
-        open: false,
-        msgOpen: false,
+        isDialogOpen: false,
       });
     }
   };
 
-  // Handle toggle between custom server and default server
   handleServeChange = event => {
```
Comment (akshatnitd) [inline src/components/Auth/SignUp/SignUp.react.js]: Remove the function. We do not have any option for custom server now. Remove related code for the Custom Server implementation

**Comment by AakashMallik** [comment]
@akshatnitd Made the changes you asked for.

File: src/redux/reducers/app.js [hunk 5]
```diff
@@ -25,6 +25,11 @@ export default handleActions(
         accessToken,
       };
     },
+    [actionTypes.APP_GET_SIGNUP](state, { payload }) {
```
Comment (akshatnitd) [inline src/redux/reducers/app.js]: I dont't think we need this reducer.

**Comment by AakashMallik** [comment]
@akshatnitd I have added a loader, removed the pop-up dialogue and also got rid of access_token. Let me know if I missed anything.

**Comment by Orbiter** [comment]
After the latest merge this has conflicts, please resolve the conflicts

**Comment by AakashMallik** [comment]
@Orbiter Thank you for pointing that out. I have dealt with the conflicts. Let me know if I missed something. :smile:

**Pull request merged**